### ISSUE TITLE
linter: possible false positive in `no-unused-var` when parameter is only used as a type predicate

### ISSUE THREAD

**Issue by uhyo** [oxc-project/oxc#22367]
### What version of Oxlint are you using?

1.64.0

### What command did you run?

`oxlint`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

Reproduces without a config file (requires `no-unused-var` to be enabled)

### What happened?

**Code:**

```ts
export function foo(a: unknown): a is string {
  return true;
}
```

oxlint flags `a` as unused despite its use in return type (`a is string`).

<details>
<summary>Error message</summary>

```
  ⚠ eslint(no-unused-vars): Parameter 'a' is declared but never used. Unused parameters should start with a '_'.
   ╭─[test.ts:2:21]
 1 │ 
 2 │ export function foo(a: unknown): a is string {
   ·                     ┬
   ·                     ╰── 'a' is declared here
 3 │   return true;
   ╰────
  help: Consider removing this parameter.
```

</details>

I am not sure if this should be treated as a bug; it's totally understandable to flag `a` as unused particularly because you can't write meaningful code without using such _a_ parameter inside function body.

I'm reporting this anyway because it diverges from ESLint behavior and I encountered this while migrating from ESLint.

Thank you for the great tool. It's fast 🙂

**Comment by camchenry** [comment]
`no-unused-vars` (base ESLint rule) doesn't seem to flag this, because it's not aware of TS syntax. However, the `@typescript-eslint/no-unused-vars` rule does seem to emit an error:

https://typescript-eslint.io/play#ts=5.9.2&fileType=.tsx&code=KYDwDg9gTgLgBAMwK4DsDGMCWEWIhACgEMAuOVAaxQgHcUBKMouTAZzlZikxQHM4A3gCg4cKMBhIouLkmABuIQF8gA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false
